app/controllers/warehouse_audit_controller.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class WarehouseAuditController < ApplicationController
+  def index
+    statistics = WarehouseAudit.where.not(user_id: nil)
+    params[:action] = nil if params[:action] == ''
+    params[:search_name] = nil if params[:search_name] == ''
+    WarehouseAudit.alphabetical
+    WarehouseAudit.search_name(params[:name])
+    WarehouseAudit.search_action(params[:action])
+    # @statistics = WarehouseAudit.search_date(params[:start_date], params[:end_date])
+    render json: statistics
```

Comment:
statistics = WarehouseAudit.where.not(user_id: nil)
                                                    .alphabetical
    statistics =  statistics.search_name(params[:name]) unless  params[:name].blank?
    statistics = statistics.search_action(params[:action]) unless (params[:action]).blank?

    render json: statistics